pex/pex.py

```diff
@@ -68,11 +68,15 @@ def _activate(self):
 
       # set up the local .pex environment
       pex_info = self._pex_info.copy()
+      pex_info_pex_path = pex_info.pex_path
+      overrides_pex_path = self._pex_info_overrides.pex_path
       pex_info.update(self._pex_info_overrides)
+      pex_info.pex_path = ":".join(filter(None, [pex_info_pex_path, overrides_pex_path]))
       self._envs.append(PEXEnvironment(self._pex, pex_info))
-
+      # by this point, pex_info.pex_path will have a single pex path merged from pex_path
+      # in PEX-INFO and pex_path set in the environment
       if pex_info.pex_path:
-        # set up other environments as specified in PEX_PATH
+        # set up other environments as specified in pex_path
         for pex_path in filter(None, pex_info.pex_path.split(os.pathsep)):
```

Comment:
how about instead of assigning to temp vars and then joining + mutating `pex_info.pex_path` here, just combining these into a new temporary `list` and then handling that directly?

also, would it make more sense to use `self._vars.PEX_PATH` as the source of the env var vs `self._pex_info_overrides.pex_path`? that's how it worked prior to your last change.

e.g.

```
# N.B. `PEX_PATH` entries written into `PEX-INFO` take precedence over those set in the environment.
combined_pex_path = self._merge_split(pex_info.pex_path, self._vars.PEX_PATH)
if combined_pex_path:
  for pex_path in combined_pex_path:
    ...
```